python/graphstorm/model/edge_gnn.py

```diff
@@ -199,19 +200,16 @@ def edge_mini_batch_gnn_predict(model, loader, return_proba=True, return_label=F
         for iter_l in range(max_num_batch):
             iter_start = time.time()
             tmp_keys = []
+            blocks = None
+            input_nodes = {}
             if iter_l < len_dataloader:
                 input_nodes, target_edge_graph, blocks = next(dataloader_iter)
                 if not isinstance(input_nodes, dict):
                     assert len(g.ntypes) == 1
                     input_nodes = {g.ntypes[0]: input_nodes}
+            if is_wholegraph():
                 tmp_keys = [ntype for ntype in g.ntypes if ntype not in input_nodes]
-                # All samples should contain all the ntypes for wholegraph compatibility
-                input_nodes.update({ntype: th.empty((0,), dtype=g.idtype) \
-                    for ntype in tmp_keys})
-            else:
-                input_nodes = {ntype: th.empty((0,), dtype=g.idtype) for ntype in g.ntypes}
-                blocks = None
-
+                prepare_for_wholegraph(g, input_nodes)
             input_feats = data.get_node_feats(input_nodes, device)
             if blocks is None:
                 continue
```

Comment:
Can also add `if is_wholegraph():` here?
as only when is_wholegraph() is True, tmp_keys will not be empty.
Otherwise, people will be confused.

Reply:
I see. I added a comment to clarify.